lib/ice_cube/parsers/ical_parser.rb

```diff
@@ -5,17 +5,19 @@ def self.schedule_from_ical(ical_string, options = {})
       ical_string.each_line do |line|
         (property, value) = line.split(':')
         (property, tzid) = property.split(';')
+        (_, time_zone) = tzid.split('=') if tzid
+
         case property
         when 'DTSTART'
-          data[:start_time] = TimeUtil.deserialize_time(value)
+          data[:start_time] = TimeUtil.deserialize_time({time: value, zone: time_zone})
         when 'DTEND'
-          data[:end_time] = TimeUtil.deserialize_time(value)
+          data[:end_time] = TimeUtil.deserialize_time({time: value, zone: time_zone})
         when 'RDATE'
           data[:rtimes] ||= []
-          data[:rtimes] += value.split(',').map { |v| TimeUtil.deserialize_time(v) }
+          data[:rtimes] += value.split(',').map { |v| TimeUtil.deserialize_time({time: v, zone: time_zone}) }
         when 'EXDATE'
           data[:extimes] ||= []
-          data[:extimes] += value.split(',').map { |v| TimeUtil.deserialize_time(v) }
+          data[:extimes] += value.split(',').map { |v| TimeUtil.deserialize_time({time: v, zone: time_zone}) }
```

Comment:
I updated the `TimeUtil.deserialize_time` to handle a hash argument where the `:zone` key might equal `nil`.

Also, while the `TimeUtil.deserialize_time` method could already handle parsing times with zone information, the ical_parser never sent zone information to the `deserialize_time()` method.